Explorer/Assets/DCL/StylizedSkybox/Scripts/SkyboxController.cs

```diff
@@ -0,0 +1,296 @@
+        }
+        else
+        {
+            //assign light to render settings
+            RenderSettings.sun = DirectionalLight;
+
+            //create animation component in runtime and assign animation clip
+            lightAnimator = DirectionalLight.gameObject.AddComponent<Animation>();
+
+            if(LightAnimation == null)
+            {
+                Debug.LogWarning("Skybox Controller: Directional Light animation has not been assigned");
+            }
+            else
+            {
+                lightAnimator.AddClip(LightAnimation,LightAnimation.name);
+            }
+        }
+
+        //setup indirect light
+        if(IndirectLight)
+        {
+            RenderSettings.ambientMode = UnityEngine.Rendering.AmbientMode.Trilight;
+        }
+
+        //setup fog
+        if(Fog)
+        {
+            RenderSettings.fog = true;
+            RenderSettings.fogMode = FogMode.Exponential;
+            RenderSettings.fogDensity = 0.001f;
+        }
+
+        isInitialized = true;
+    }
+
+    void Start()
+    {
+        SetTime(SecondsInDay/2);
+        if(PlayOnStart) { Play();} else {Pause();}
+    }
+
+    void Update()
+    {
+        if (!isInitialized)
+            return;
+
+        //update natural and relative time
+        if(!pause)
+        {
+            float deltaTime = (Time.deltaTime * Speed);
+            NaturalTime += deltaTime;
+            NormalizedTime +=  deltaTime / SecondsInDay;
+        }
+
+        //loops time at the end of the cycle
+        if (NaturalTime >= SecondsInDay)
+        {
+            NaturalTime = NormalizedTime = 0;
+        }
+
+        //update skybox only after certain time
+        sinceLastRefresh += Time.deltaTime;
+        if(sinceLastRefresh >= RefreshTime)
+        {
+            UpdateSkybox();
+            sinceLastRefresh = 0;
+        }
+
+        UpdateTimeUI();
+    }
+
+    /// <summary>
+    /// Auxiliary method for the Inspector to Pause the cycle
+    /// </summary>
+    public void Pause()
+    {
+        pause = true;
+    }
+
+    /// <summary>
+    /// Auxiliary method for the Inspector to Play the cycle
+    /// </summary>
+    public void Play()
+    {
+        pause = false;
+    }
+
+    /// <summary>
+    /// Set the material of the skybox and modify the Render Settings
+    /// </summary>
+    /// <param name="skyboxMaterial"></param>
+    public void SetSkyboxMaterial(Material skyboxMaterial)
+    {
+        this.skyboxMaterial = skyboxMaterial;
+        RenderSettings.skybox = this.skyboxMaterial;
+    }
+
+    /// <summary>
+    /// Sets the time of the skybox to an specific second
+    /// </summary>
+    /// <param name="seconds"></param>
+    public void SetTime(float seconds)
+    {
+        NaturalTime = seconds;
+        NormalizedTime = NaturalTime / SecondsInDay;
+    }
+
+    /// <summary>
+    /// Auxiliary function to returnt the normalized time in HH:MM:SS
+    /// </summary>
+    public string GetFormatedTime()
```

Comment:
I think this can be private

Reply:
This information can be displayed by other UIs without accessing the Normalized or Natural time